File: lc171.py
class Solution(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        LETTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        
        dct_temp = {}
        
        for i in range(26):
            dct_temp[LETTERS[i]] = i + 1
    
        res = 0
        for c in s:
            res = 26*res + dct_temp[c]
        return res

File: lc171.java
class Solution {
    public int titleToNumber(String s) {
        char[] c = s.toCharArray();
        int res = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            res += (c[i] - 64) * (int) Math.pow(26, s.length() - i - 1);
        }
        return res;
    }
}